gnucash/register/register-gnome/completioncell-gnome.c

```diff
@@ -0,0 +1,1010 @@
+{
+    char *normalized, *folded;
+    g_return_val_if_fail (utf8_string && *utf8_string, NULL);
+
+    normalized = normalize (utf8_string);
+    if (!normalized)
+        return NULL;
+
+    folded = casefold (normalized);
+    g_free (normalized);
+    return folded;
+}
+
+static void
+add_item_anywhere (gpointer key, gpointer value, gpointer user_data)
+{
+    PopBox* box = user_data;
+    gchar *hash_entry = g_strdup (key);
+
+    box->using_anywhere = TRUE;
+
+    if (hash_entry && *hash_entry)
+    {
+        gnc_utf8_strip_invalid_and_controls (hash_entry);
+        gchar *key_norm      = normalize (hash_entry);
+        gchar *key_norm_fold = casefold (key_norm);
+
+        if (g_strstr_len (key_norm_fold, -1, box->newval) != NULL)
+            list_store_append (box->item_store, hash_entry, key_norm, GPOINTER_TO_INT(value));
+
+        g_free (key_norm_fold);
+        g_free (key_norm);
+    }
+    g_free (hash_entry);
+}
+
+static void
+add_item_start (gpointer key, gpointer value, gpointer user_data)
+{
+    PopBox* box = user_data;
+    gchar *hash_entry = g_strdup (key);
+
+    if (hash_entry && *hash_entry)
+    {
+        gnc_utf8_strip_invalid_and_controls (hash_entry);
+        gchar *key_norm      = normalize (hash_entry);
+        gchar *key_norm_fold = casefold (key_norm);
+
+        if (g_str_has_prefix (key_norm_fold, box->newval))
+            list_store_append (box->item_store, hash_entry, key_norm, GPOINTER_TO_INT(value));
+
+        g_free (key_norm_fold);
+        g_free (key_norm);
+    }
+    g_free (hash_entry);
+}
+
+static void
+populate_list_store (CompletionCell* cell, const gchar* str)
+{
+    PopBox* box = cell->cell.gui_private;
+
+    clock_t start = clock();
+g_print("### pop store start\n");
+
+    box->in_list_select = FALSE;
+
+    if (box->stop_searching)
+        return;
+
+    if (str && *str)
+        box->newval = normalize_and_fold ((gchar*)str);
+    else
+        return;
+
+//FIXME this would start filtering on second character, if not required remove and tooltip
+//    if (!box->filter_at_start && g_utf8_strlen (str, -1) < 2)
+//        return;
+
+    // disconnect list store from tree view
+    box->item_store = gnc_item_list_drop_store (box->item_list);
+
+    block_list_signals (cell);
+
+    if (box->sort_enabled) // if sorting, disable it
+        gnc_item_list_set_sort_enabled (box->item_list, FALSE, box->sort_column_id);
+
+    gtk_list_store_clear (box->item_store);
+
+    // add to the list store
+    if (box->filter_at_start && !box->using_anywhere)
+        g_hash_table_foreach (box->item_hash, add_item_start, box);
+    else
+        g_hash_table_foreach (box->item_hash, add_item_anywhere, box);
+
+    // if searched from start with no results, try anywhere.
+    if (box->filter_at_start && !box->using_anywhere &&
+        gtk_tree_model_iter_n_children (GTK_TREE_MODEL(box->item_store), NULL) == 0)
+    {
+        g_hash_table_foreach (box->item_hash, add_item_anywhere, box);
+    }
```

Comment:
This code is likely to  change anyway, but I would have suggested to move this bit inside the earlier if clause above...

Reply:
Not required as there is only one way of filtering now.